### DISCUSSION TITLE
User is null when calling supabaseClient.auth.getUser() in edge function on local development

### DISCUSSION THREAD

**Discussion by jessecarrigan** [supabase/supabase#22169] [Questions]
Hi there! I've been following along with the docs [here](https://supabase.com/docs/guides/functions/auth), but am stumped on this issue. I'm on a local dev instance of Supabase. In my edge function, I am attempting to get the currently logged-in user by calling auth.getUser() on the Supabase client. I am trying to get the user id in order to store some values along with the logged-in id, which should be a user that is authenticated through Supabase Auth on the front-end. However, the user comes back as null in the edge function. I checked that the authorization header is being set correctly where the React app on the browser is calling the edge function and the client is being initialized correctly, but the user is always null despite getting the expected value for the Authorization header.

I've seen a number of similar discussions but none of the existing threads have shed any light on the issue. Here is the relevant code in the function, leaving out some of the specific details of what should be saved to the database:

```
import { corsHeaders } from "../_shared/cors.ts"
import { createClient } from 'https://esm.sh/@supabase/supabase-js@2'

console.log("Starting new verification...")

Deno.serve(async (req) => {
  // This is needed if you're planning to invoke your function from a browser.
  if (req.method === 'OPTIONS') {
    return new Response('ok', { headers: corsHeaders })
  }

  const supabaseClient = createClient(
    Deno.env.get('SUPABASE_URL') ?? '',
    Deno.env.get('SUPABASE_ANON_KEY') ?? '',
    { global: { headers: { Authorization: req.headers.get('Authorization')! } },
      auth: { persistSession: false } }
  )

  // Get the session or user object
  const { data } = await supabaseClient.auth.getUser()
  const user = data.user

  // Use user id to store data here...

  } catch (error) {
    console.error(error)
    return new Response(
      JSON.stringify({ data: "An error occurred" }),
      { headers: { ...corsHeaders, "Content-Type": "application/json" } },
    )
  }
  return new Response(
    JSON.stringify({ data: transcript }),
    { headers: { ...corsHeaders, "Content-Type": "application/json" } },
  )
})
```

Thanks for any recommendations!

**Comment by GaryAustin1** [comment]
Forcing the authorization header to the user jwt does not set a session in the client.  You can't do getUser() without that session.  You need to do getUser(jwt) with the jwt extracted from the authorization header.

This has a user's code that extracts the jwt: https://github.com/orgs/supabase/discussions/19621

**Comment by gittagberdi** [comment]
What helped me is to set NEXT_SUPABASE_URL = 'http://host.docker.internal:54321'. According to Deepseek, "The root cause is that your Deno edge function is running inside a Docker container (part of Supabase's local setup), while your Supabase auth service is exposed on the host machine's 127.0.0.1:54321. Containers can't access the host via 127.0.0.1 directly."

**Comment by jt6677** [comment]
> Hi there! I've been following along with the docs [here](https://supabase.com/docs/guides/functions/auth), but am stumped on this issue. I'm on a local dev instance of Supabase. In my edge function, I am attempting to get the currently logged-in user by calling auth.getUser() on the Supabase client. I am trying to get the user id in order to store some values along with the logged-in id, which should be a user that is authenticated through Supabase Auth on the front-end. However, the user comes back as null in the edge function. I checked that the authorization header is being set correctly where the React app on the browser is calling the edge function and the client is being initialized correctly, but the user is always null despite getting the expected value for the Authorization header.
> 
> I've seen a number of similar discussions but none of the existing threads have shed any light on the issue. Here is the relevant code in the function, leaving out some of the specific details of what should be saved to the database:
> 
> ```
> import { corsHeaders } from "../_shared/cors.ts"
> import { createClient } from 'https://esm.sh/@supabase/supabase-js@2'
> 
> console.log("Starting new verification...")
> 
> Deno.serve(async (req) => {
>   // This is needed if you're planning to invoke your function from a browser.
>   if (req.method === 'OPTIONS') {
>     return new Response('ok', { headers: corsHeaders })
>   }
> 
>   const supabaseClient = createClient(
>     Deno.env.get('SUPABASE_URL') ?? '',
>     Deno.env.get('SUPABASE_ANON_KEY') ?? '',
>     { global: { headers: { Authorization: req.headers.get('Authorization')! } },
>       auth: { persistSession: false } }
>   )
> 
>   // Get the session or user object
>   const { data } = await supabaseClient.auth.getUser()
>   const user = data.user
> 
>   // Use user id to store data here...
> 
>   } catch (error) {
>     console.error(error)
>     return new Response(
>       JSON.stringify({ data: "An error occurred" }),
>       { headers: { ...corsHeaders, "Content-Type": "application/json" } },
>     )
>   }
>   return new Response(
>     JSON.stringify({ data: transcript }),
>     { headers: { ...corsHeaders, "Content-Type": "application/json" } },
>   )
> })
> ```
> 
> Thanks for any recommendations!

  const { data } = await supabaseClient.auth.getUser()   this used to work. As 1-25-2025, it does not work any more .  You need to pass in the token wiht 'Bear' removed.  not sure what changed.

**Reply by eatrisno** [reply]
any solution for this issue?

**Reply by DWboutin** [reply]
I'm having the same problem
